src/DotLiquid/Tokenizer.cs

```diff
@@ -90,13 +99,68 @@ internal static List<string> Tokenize(string source)
             return tokens;
         }
 
+        /// <summary>
+        /// Enumerates over a variable sequence in dotted or bracket notation
+        /// </summary>
+        /// <param name="source">The Liquid Variable string</param>
+        /// <returns></returns>
+        /// <exception cref="SyntaxException"></exception>
+        internal static IEnumerator<string> GetVariableEnumerator(string source)
+        {
+            if (string.IsNullOrEmpty(source))
+                yield break;
+
+            using (var markupEnumerator = new CharEnumerator(source))
+            {
+                while (markupEnumerator.HasNext())
+                {
+                    var isComplete = false;
+                    var sb = new StringBuilder();
```

Comment:
nit: consider more description variable name than `sb` - what is the string builder's purpose or content?

Reply:
Will fix this in this method, but as a parameter for the other methods it is consistent with Microsoft's own StringWriter constructor.

Reply:
We haven't always been very good at following our own best practices guidance. And old habits are pretty resilient to change :-)